src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -58,184 +61,336 @@ using namespace chip;
+{
+    sCommissionerSecureSessionHandle = secureSessionHandle;
+}
 
-            sNetworks[i].mNetworkType = NetworkType::kThread;
-            sNetworks[i].mEnabled     = false;
+void EncodeAndSendEnableNetworkResponse(CHIP_ERROR err)
+{
+    CHIP_ERROR processError                  = CHIP_NO_ERROR;
+    chip::app::CommandSender * commandSender = nullptr;
+    EmberAfNetworkCommissioningError commissioningError =
+        (err == CHIP_NO_ERROR ? EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS
+                              : EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR);
+    NodeId commissionerNodeId              = sCommissionerSecureSessionHandle.GetPeerNodeId();
+    Transport::AdminId commissionerAdminId = sCommissionerSecureSessionHandle.GetAdminId();
+
+    // From the Network Commissioning cluster spec, we MAY put response command in seperate exchange, and may have a StatusCode
+    // before the ResponseCommand.
+    app::CommandPathParams cmdParams = { 0, /* group id */ 0, ZCL_NETWORK_COMMISSIONING_CLUSTER_ID,
+                                         ZCL_ENABLE_NETWORK_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+    TLV::TLVWriter * writer          = nullptr;
+    SuccessOrExit(processError = chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&commandSender));
+    SuccessOrExit(processError = commandSender->PrepareCommand(&cmdParams));
+    writer = commandSender->GetCommandDataElementTLVWriter();
+    SuccessOrExit(processError = writer->Put(TLV::ContextTag(0), static_cast<uint32_t>(commissioningError)));
+    SuccessOrExit(processError = writer->PutString(TLV::ContextTag(1), ""));
+    SuccessOrExit(processError = commandSender->FinishCommand());
+    SuccessOrExit(processError = commandSender->SendCommandRequest(commissionerNodeId, commissionerAdminId));
+
+    // Interaction model engine will manage the state of command sender when successfully send this command, so we can safely delete
+    // this.
+    commandSender = nullptr;
 
-            err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS;
-            break;
-        }
+exit:
+    if (commandSender != nullptr)
+    {
+        commandSender->Shutdown();
     }
+    ChipLogError(Zcl, "Failed to send response command to commissioner: %d", processError);
+}
 
-exit:
-    // TODO: We should encode response command here.
+} // namespace Internal
 
-    ChipLogDetail(Zcl, "AddThreadNetwork: %d", err);
-    return err;
+// Actual command handlers.
+
+void OnAddThreadNetworkCommandCallbackInternal(app::Command * apCommandObj, EndpointId endpointId, ByteSpan operationalDataset,
+                                               uint64_t breadcrumb, uint32_t timeoutMs)
+{
+    CHIP_ERROR encodingError             = CHIP_NO_ERROR;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
```

Comment:
I'm a little confused.  If not `CHIP_DEVICE_CONFIG_ENABLE_THREAD`, why does this function exist at all?

Reply:
Moved CHIP_DEVICE_CONFIG_ENABLE_THREAD check and CHIP_DEVICE_CONFIG_ENABLE_WPA check to remove the corresponding functions if they does not exists.